components/util/default/maybe.mjs

```diff
@@ -0,0 +1,8 @@
+export const fromMaybe = (maybe, recovery, transform) =>
+  maybe === null ? recovery : transform(maybe);
+
+export const mapMaybe = (maybe, transform) =>
+  maybe === null ? null : transform(maybe);
+
+export const mapMaybeAsync = async (maybe, transformAsync) =>
+  maybe === null ? null : await transformAsync(maybe);
```

Comment:
Well you are right. But when I code, I'm always running a type checker in my head. And I hate mixing promise (monadic) values with pure value. But look, I avoid the polemic by simply removing it since it is deadcode anyhow :)

Reply:
> I'm always running a type checker in my head

Perhaps you should consider using an actual typechecker with TypeScript...